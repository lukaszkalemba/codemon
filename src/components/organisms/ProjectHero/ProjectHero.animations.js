const description = {
  variants: {
    animate: {
      opacity: 0.8,
      x: 0,
      transition: { duration: 0.8, delay: 0.25 },
    },
    initial: { opacity: 0, x: -30 },
  },
  initial: 'initial',
  animate: 'animate',
};

const background = {
  variants: {
    animate: {
      opacity: 0.8,
      x: 0,
      transition: { duration: 0.8, delay: 0.25 },
    },
    initial: { opacity: 0, x: 100 },
  },
  initial: 'initial',
  animate: 'animate',
};

const scroll = {
  variants: {
    initial: { opacity: 0, y: -50 },
    animate: { opacity: 1, y: 0, transition: { duration: 0.75, delay: 0.25 } },
  },
  initial: 'initial',
  animate: 'animate',
};

export default { description, background, scroll };
