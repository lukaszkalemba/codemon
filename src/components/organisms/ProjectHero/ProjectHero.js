import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { setHeightOnMobile } from 'helpers/setHeightOnMobile';
import Container from 'components/particles/Container';
import ProjectHeroContent from 'components/molecules/ProjectHeroContent';
import S from './ProjectHero.styles';
import animations from './ProjectHero.animations';

const ProjectHero = ({ title, description, image }) => {
  useEffect(() => {
    setHeightOnMobile();
  }, []);

  return (
    <S.Wrapper>
      <S.HeadingWrapper>
        <Container axis="both" doubling="x">
          <ProjectHeroContent
            title={title}
            description={description}
            animation={animations.description}
          />
        </Container>
      </S.HeadingWrapper>
      <S.BackgroundWrapper {...animations.background}>
        <S.Background fluid={image}>
          <S.Overlay />
        </S.Background>
      </S.BackgroundWrapper>
    </S.Wrapper>
  );
};

ProjectHero.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.shape({
    fluid: PropTypes.shape({}),
  }).isRequired,
};

export default ProjectHero;
