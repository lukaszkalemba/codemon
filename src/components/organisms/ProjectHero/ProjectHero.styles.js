import styled from 'styled-components';
import { motion } from 'framer-motion';
import BackgroundImage from 'gatsby-background-image';

const Wrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-template-rows: repeat(1, 1fr);
  padding: 10px 0;
  height: calc(var(--vh, 1vh) * 88);

  ${({ theme }) => theme.media.laptop`
    height: 85vh;
  `}
`;

const HeadingWrapper = styled.div`
  grid-column: 1 / 6;
  grid-row: 1 / 2;
  position: relative;
  z-index: ${({ theme }) => theme.zindex.level1};

  ${({ theme }) => theme.media.laptop`
    grid-column: 1 / 5;
  `}

  ${({ theme }) => theme.media.desktop`
    grid-column: 1 / 4;
  `}
`;

const BackgroundWrapper = styled(motion.div)`
  position: relative;
  grid-column: 3 / 6;
  grid-row: 1 / 2;
  height: 100%;
`;

const Background = styled(BackgroundImage)`
  position: relative;
  grid-column: 3 / 6;
  grid-row: 1 / 2;
  height: 100%;
`;

const Overlay = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.95;
  background-color: ${({ theme }) => theme.color.dark};
`;

export default {
  Wrapper,
  HeadingWrapper,
  BackgroundWrapper,
  Background,
  Overlay,
};
