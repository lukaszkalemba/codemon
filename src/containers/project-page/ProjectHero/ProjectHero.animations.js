const descriptionVariants = {
  animate: {
    opacity: 0.8,
    x: 0,
    transition: { duration: 0.8, delay: 0.25 },
  },
  initial: { opacity: 0, x: -30 },
};

const backgroundVariants = {
  animate: {
    opacity: 0.8,
    x: 0,
    transition: { duration: 0.8, delay: 0.25 },
  },
  initial: { opacity: 0, x: 100 },
};

export default { descriptionVariants, backgroundVariants };
