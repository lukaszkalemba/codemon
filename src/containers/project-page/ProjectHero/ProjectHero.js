import React from 'react';
import PropTypes from 'prop-types';
import Image from 'gatsby-image';
import Container from 'components/common/Container';
import Heading from 'components/common/Heading';
import S from './ProjectHero.styles';

const ProjectHero = ({ title, description, image }) => {
  return (
    <S.Wrapper>
      <Container axis="both">
        <Heading>{title}</Heading>
        <S.Description>{description}</S.Description>
      </Container>
      <S.ImageWrapper>
        <Image fluid={image} />
        <S.Overlay />
      </S.ImageWrapper>
    </S.Wrapper>
  );
};

ProjectHero.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.shape({
    fluid: PropTypes.object,
  }).isRequired,
};

export default ProjectHero;
