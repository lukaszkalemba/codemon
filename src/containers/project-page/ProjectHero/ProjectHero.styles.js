import styled from 'styled-components';
import { motion } from 'framer-motion';
import BackgroundImage from 'gatsby-background-image';

const Wrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-template-rows: repeat(1, 1fr);
`;

const HeadingWrapper = styled.div`
  grid-column: 1 / 6;
  grid-row: 1 / 2;
  position: relative;
  z-index: ${({ theme }) => theme.zindex.positive1};

  ${({ theme }) => theme.media.laptop`
    grid-column: 1 / 5;
  `}

  ${({ theme }) => theme.media.desktop`
    grid-column: 1 / 4;
  `}
`;

const Description = styled(motion.p)`
  font-size: ${({ theme }) => theme.font.size.xxs};
  opacity: 0.8;
  line-height: 1.75em;

  ${({ theme }) => theme.media.tablet`
    font-size: ${theme.font.size.xs};
  `}
`;

const BackgroundWrapper = styled(motion.div)`
  position: relative;
  grid-column: 3 / 6;
  grid-row: 1 / 2;
  height: 700px;
`;

const Background = styled(BackgroundImage)`
  position: relative;
  grid-column: 3 / 6;
  grid-row: 1 / 2;
  height: 700px;
`;

const Overlay = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.95;
  background-color: ${({ theme }) => theme.color.dark};
`;

export default {
  Wrapper,
  HeadingWrapper,
  Description,
  BackgroundWrapper,
  Background,
  Overlay,
};
