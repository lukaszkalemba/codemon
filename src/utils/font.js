export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
