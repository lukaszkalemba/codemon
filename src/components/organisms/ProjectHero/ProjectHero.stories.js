import React from 'react';
import { text, withKnobs } from '@storybook/addon-knobs';
import ProjectHero from './ProjectHero';

export const projectHero = () => (
  <ProjectHero
    title={text('Project Hero Title', 'Lorem ipsum.')}
    description={text(
      'Project Hero Description',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Enim incidunt debitis consequatur provident veniam labore, vel voluptatibus tenetur unde assumenda similique nam voluptate repellendus, recusandae rerum. Officia architecto laudantium asperiores!'
    )}
    image={{ fluid: {} }}
  />
);

export default {
  component: ProjectHero,
  decorators: [withKnobs],
  title: 'Organisms|ProjectHero',
};
