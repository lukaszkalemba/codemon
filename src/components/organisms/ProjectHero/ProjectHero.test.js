import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import ProjectHero from './ProjectHero';

describe('<ProjectHero />', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = render(<ProjectHero />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
