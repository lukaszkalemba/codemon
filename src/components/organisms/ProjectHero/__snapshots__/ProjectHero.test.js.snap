// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectHero /> renders correctly and matches snapshot 1`] = `
<div
  class="sc-AxhUy hwCBdc"
>
  <div
    class="sc-AxgMl gXArbd"
  >
    <div
      class="sc-AxjAm jLocZ"
    >
      <h2
        class="sc-AxiKw kKlpRG"
        data-testid="heading"
        style="opacity: 0; transform: translateX(40px) translateZ(0);"
      />
      <p
        class="sc-AxhCb eyrdko"
        data-testid="description"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      />
    </div>
  </div>
  <div
    class="sc-AxheI jxxMsa"
    style="opacity: 0; transform: translateX(100px) translateZ(0);"
  />
</div>
`;
